sanic/request.py

```diff
@@ -312,13 +312,13 @@ def parse_multipart_form(body, boundary):
                 file_name = form_parameters.get('filename')
                 field_name = form_parameters.get('name')
             elif form_header_field == 'content-type':
-                content_type = form_header_value
                 content_charset = form_parameters.get('charset', 'utf-8')
 
         if field_name:
             post_data = form_part[line_index:-4]
             if file_name:
-                form_file = File(type=content_type,
+                file_type = guess_type(file_name)[0] or 'text/plain'
```

Comment:
I think this should still use `form_header_value` when present, e.g.: `file_type = form_header_value  or guess_type(file_name)[0] or 'text/plain'`

Reply:
How would that work exactly?

`Content-Disposition` is the only header that we generate a `File` on. `Content-Type` is a whole other header.

Are you proposing bringing back the `content_type` variable and storing in that?

---

What about the cases where `Content-Type` comes after `Content-Disposition`? It'll not use that Content-Type in such cases. Old code would also cause issues with that, sure, but should we really reintroduce a bug?

---

Additionally, bringing back `Content-Type` brings 2 issues:

- Users can fake the type of files, which might be undesirable (currently they need to change the extension to fake it, which is relatively more secure in cases where a whitelist is introduced (as people can't just upload their exes and send a `Content-Type` of say, `image/png`, and in the current state users would need to rename the file to run them as executables))

- `Content-Type` doesn't necessarily show the type of file. Originally when we had this issue the code wasn't returning `text/plain` (which is default) but `application/octet-stream`, as that's what the request had as `Content-Type`

Reply:
I'm a little confused here. `Content-Disposition` is a browser related header, it only provides information regarding how the browser should handle with the content it receives in the body. From the [MDN](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Disposition):

> In a regular HTTP response, the Content-Disposition response header is a header indicating if the content is expected to be displayed inline in the browser, that is, as a Web page or as part of a Web page, or as an attachment, that is downloaded and saved locally.

AFAIK, `Content-Type` is almost a must have to differentiate what's the kind of data is being sent over the body. `application/octet-stream` is the _de facto_ value for "all I know is that this is binary data", but this can be further refined to meet some end. See [this answer](https://stackoverflow.com/a/20509354) on SO regarding a similar topic.

Perhaps we should get the RFC and strict to the expected behavior and, if one needs, just override it, for request or response.

Reply:
I mentioned `Content-Disposition` as we only create a `File` object when we're processing that (so checking `form_header_value` before creating it would have `Content-Disposition` info).

Also it's more than that, it also provides info about filename and such, which is why it's used by sanic.

---

`application/octet-stream` is all we get, no matter type of file. That's not helpful.

---

Which RFC would be relevant to this? I'm willing to check it out.

Reply:
I know that `Content-Disposition` may provide a fine grained control over received files, but it is not something we'll always have at our disposal (and we really should take that under consideration). If that's the case, what do you suggest could be the default behavior?

You can check RFC 1867 (obsoleted), 2388 (proposed, obsoleted), 2854 (informational), 7578 (proposed).

Reply:
I don't know what `Content-Disposition` could be replaced with. I suppose in rare cases where the header is not present, the file name and type could be left as an empty str or just as null. Though I'm not quite sure how a file's presence would be checked in the first place.

---

To be clear, when I say `application/octet-stream is all we get, no matter type of file. That's not helpful.`, I mean we as in @elixire, not sanic (though we do use sanic in prod and had this behavior).

The behavior might be different under different environments and different clients, however this was the case for us.

---

I didn't get a chance to read the RFCs but I'll try to read them tomorrow and might add a second comment.